activerecord/test/cases/insert_all_test.rb

```diff
@@ -78,6 +78,16 @@ def test_insert_all_raises_on_duplicate_records
     end
   end
 
+  def test_insert_all_raises_on_duplicate_records_when_unqiue_by_is_provided
+    assert_raise ActiveRecord::RecordNotUnique do
+      Cart.insert_all! [
+        { id: 1, shop_id: 1, title: "My cart" },
+        { id: 2, shop_id: 1, title: "My another cart" },
+        { id: 1, shop_id: 1, title: "My another cart" },
+      ], unique_by: [:shop_id, :id]
+    end
+  end
```

Comment:
The counterparts are here that doesn't raise an error

https://github.com/rails/rails/blob/6cda7694098703bc4a3c06328d0c00b46a8a2c42/activerecord/test/cases/insert_all_test.rb#L121-L142